app/controllers/users_controller.rb

```diff
@@ -17,7 +17,10 @@ def create
   end
 
   def show
-
+    render file: "/public/404" unless current_user
+    if current_user
+      @user = User.find(session[:user_id])
+    end
```

Comment:
Solves visitiors seeing profile pages. Now mush be current_user. Also defined @user for user show pages so that we can use the variable instead of `session[:user]` each time.